src/components/MoneyReportHeader.tsx

```diff
@@ -235,6 +235,7 @@ function MoneyReportHeader({
     const isPayAtEndExpense = isPayAtEndExpenseTransactionUtils(transaction);
     const isArchivedReport = useReportIsArchived(moneyRequestReport?.reportID);
     const isChatReportArchived = useReportIsArchived(chatReport?.reportID);
+    const [allSnapshots] = useOnyx(ONYXKEYS.COLLECTION.SNAPSHOT, {canBeMissing: true});
```

Comment:
Can we pass the specific snapshot key we want here? e.g.

```
const currentSearchQueryJSON = getCurrentSearchQueryJSON();
const hash = currentSearchQueryJSON?.hash ?? CONST.DEFAULT_NUMBER_ID;
const [allSnapshots] = useOnyx(`${ONYXKEYS.COLLECTION.SNAPSHOT}${hash}`, {canBeMissing: true});
```

We'd also be able to get rid of the code above in each of the functions, e.g. approveMoneyRequest, etc